types/orchestration.d.ts

```diff
@@ -205,29 +166,22 @@ export declare class DurableOrchestrationContext {
      * Schedules an orchestration function named `name` for execution.
      *
      * @param name The name of the orchestrator function to call.
-     * @param input The JSON-serializable input to pass to the orchestrator
-     * function.
-     * @param instanceId A unique ID to use for the sub-orchestration instance.
-     * If `instanceId` is not specified, the extension will generate an id in
-     * the format `<calling orchestrator instance ID>:<#>`
+     * @param options optional object to control the scheduled orchestrator (e.g provide input, instanceID)
      */
-    callSubOrchestrator(name: string, input?: unknown, instanceId?: string): Task;
+    callSubOrchestrator(name: string, options?: CallSubOrchestratorOptions): Task;
 
     /**
      * Schedules an orchestrator function named `name` for execution with retry
      * options.
      *
      * @param name The name of the orchestrator function to call.
      * @param retryOptions The retry options for the orchestrator function.
-     * @param input The JSON-serializable input to pass to the orchestrator
-     * function.
-     * @param instanceId A unique ID to use for the sub-orchestration instance.
+     * @param options optional object to control the scheduled orchestrator (e.g provide input, instanceID)
      */
     callSubOrchestratorWithRetry(
         name: string,
         retryOptions: RetryOptions,
```

Comment:
It's a bit weird to have `retryOptions` _and_ `options`. Any chance we can combine them? I know it's a bit complicated because `RetryOptions` is a class...

Reply:
I'd be ok just put the retryOptions inside the new options object.

Reply:
I'd be okay with having the `retryOptions` under the `options` object too, but in that case, `callSubOrchestrator()` and `callSubOrchestratorWithRetry()` start to look awfully similar. How do we feel about combining these two APIs in this case? I discussed this with @davidmrdavid offline too. Ofc this can still be done in a separate PR.